### DISCUSSION TITLE
add spool selected in HA

### DISCUSSION THREAD

**Discussion by Eloura74** [Disane87/spoolman-homeassistant#75] [Ideas]
I made a direct integration with Moonraker via the RESTful sensor, to display the reels selected in klipper directly in Home assistant, then I created images with different colors which function as layers which are superimposed to continue my floorplan
![image](https://github.com/Disane87/spoolman-homeassistant/assets/95535946/b3a68d65-deb4-4bd5-aff2-f7df35418b1a)
![image](https://github.com/Disane87/spoolman-homeassistant/assets/95535946/f7bda4a7-97bb-4c31-8859-9327016fc015)
as well as put the different coils in a honeycomb
![image](https://github.com/Disane87/spoolman-homeassistant/assets/95535946/a7ebe0b5-dfc6-41a6-9afc-bda5d079c4ed)

here's how:
#################
color:
#################
- sensor: 
  - platform: rest
    name: Klipper Active Filament Spool
    resource: http://192.168.1.130:7125/server/spoolman/spool_id
    method: GET
    value_template: '{{ value_json.result.spool_id }}'  # Mise à jour ici
    scan_interval: 60
    headers:
      Content-Type: application/json
###################
honeycomb:
###################
  - type: custom:button-card
    name: Honey Spoolman
    styles:
      entity_picture:
        - height: 100%
        - width: 40%
        - transform: scaleX(0.6) scaleY(0.7) rotateX(4deg)
      card:
        - filter: opacity(100%)
        - box-shadow: unset
        - background-color: unset
        - background: none
        - border: none
    style:
      top: 15%
      left: 8%
    entity: light.canape
    entity_picture: /local/icone/spool12.png
    show_entity_picture: true
    show_name: false
    tap_action:
      action: call-service
      service: honeycomb
      service_data:
        size: 320
        active: true
        buttons:
          - type: custom:button-card
            entity: sensor.sunlu_black_pla
            name: Sunlu PLA
            show_name: true
            show_state: true
            show_icon: true
            icon: bha:sonarr
            styles:
              card:
                - width: 100%
                - background: linear-gradient(to right, grey, darkgrey)
              name:
                - font-size: 13px
                - color: white
                - position: absolute
                - top: 12%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 2px 2px black
              state:
                - font-size: 18px
                - color: black
                - position: absolute
                - top: 60%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 1px 1px white
              icon:
                - color: >-
                    [[[ return
                    `#${states['sensor.sunlu_black_pla'].attributes.filament_color_hex}`;
                    ]]]
            tap_action:
              action: more-info
          - type: custom:button-card
            entity: sensor.sunlu_blanc_pla
            name: Sunlu PLA
            show_name: true
            show_state: true
            show_icon: true
            icon: bha:sonarr
            styles:
              card:
                - width: 100%
                - background: linear-gradient(to right, grey, darkgrey)
              name:
                - font-size: 13px
                - color: white
                - position: absolute
                - top: 12%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 2px 2px black
              state:
                - font-size: 18px
                - color: black
                - position: absolute
                - top: 60%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 1px 1px white
              icon:
                - color: >-
                    [[[ return
                    `#${states['sensor.sunlu_blanc_pla'].attributes.filament_color_hex}`;
                    ]]]
            tap_action:
              action: more-info
          - type: custom:button-card
            entity: sensor.sunlu_bleu_pla
            name: Sunlu PLA
            show_name: true
            show_state: true
            show_icon: true
            icon: bha:sonarr
            styles:
              card:
                - width: 100%
                - background: linear-gradient(to right, grey, darkgrey)
              name:
                - font-size: 13px
                - color: white
                - position: absolute
                - top: 12%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 2px 2px black
              state:
                - font-size: 18px
                - color: black
                - position: absolute
                - top: 60%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 1px 1px white
              icon:
                - color: >-
                    [[[ return
                    `#${states['sensor.sunlu_bleu_pla'].attributes.filament_color_hex}`;
                    ]]]
            tap_action:
              action: more-info
          - type: custom:button-card
            entity: sensor.sunlu_gris_pla
            name: Sunlu PLA
            show_name: true
            show_state: true
            show_icon: true
            icon: bha:sonarr
            styles:
              card:
                - width: 100%
                - background: linear-gradient(to right, grey, darkgrey)
              name:
                - font-size: 13px
                - color: white
                - position: absolute
                - top: 12%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 2px 2px black
              state:
                - font-size: 18px
                - color: black
                - position: absolute
                - top: 60%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 1px 1px white
              icon:
                - color: >-
                    [[[ return
                    `#${states['sensor.sunlu_gris_pla'].attributes.filament_color_hex}`;
                    ]]]
            tap_action:
              action: more-info
          - type: custom:button-card
            entity: sensor.sunlu_rouge_pla
            name: Sunlu PLA
            show_name: true
            show_state: true
            show_icon: true
            icon: bha:sonarr
            styles:
              card:
                - width: 100%
                - background: linear-gradient(to right, grey, darkgrey)
              name:
                - font-size: 13px
                - color: white
                - position: absolute
                - top: 12%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 2px 2px black
              state:
                - font-size: 18px
                - color: black
                - position: absolute
                - top: 60%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 1px 1px white
              icon:
                - color: >-
                    [[[ return
                    `#${states['sensor.sunlu_rouge_pla'].attributes.filament_color_hex}`;
                    ]]]
            tap_action:
              action: more-info
          - type: custom:button-card
            entity: sensor.sunlu_multi_bleu_pla
            name: Sunlu PLA
            show_name: true
            show_state: true
            show_icon: true
            icon: bha:sonarr
            styles:
              card:
                - width: 100%
                - background: linear-gradient(to right, grey, darkgrey)
              name:
                - font-size: 13px
                - color: white
                - position: absolute
                - top: 12%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 2px 2px black
              state:
                - font-size: 18px
                - color: black
                - position: absolute
                - top: 60%
                - left: 50%
                - transform: translateX(-50%)
                - text-shadow: 1px 1px white
              icon:
                - color: >-
                    [[[ return
                    `#${states['sensor.sunlu_multi_bleu_pla'].attributes.filament_color_hex}`;
                    ]]]
            tap_action:
              action: more-info

**Comment by Disane87** [comment]
Thats nice, thank you!

**Comment by Disane87** [comment]
@Eloura74 I guess you can ditch the manual rest sensor since I support setting Klipper url to get the active spool and save that particular information within the appropiate spool. See #86 for more.

**Reply by Eloura74** [reply]
I'm sorry I didn't understand, being French and the translation being... very bad, I didn't understand.

**Reply by Disane87** [reply]
@Eloura74 Je pense que vous pouvez abandonner le capteur de repos manuel puisque je soutiens le réglage de l'url de Klipper pour obtenir la bobine active et sauvegarder cette information particulière dans la bobine appropriée. Voir #86 pour plus d'informations.

(Translated by deepl)

**Reply by Eloura74** [reply]
ok thank you very much, and thank you again for your work

**Reply by Eloura74** [reply]
I have another question, I am currently in the process of making my personal voice assistant, with x integration ranging from opening web pages, to applications, to gpt chat, to video and others, I managed to do  operate the klipper macros by voice command like home, heat, etc., do you have any idea how to macro call this or that coil, eject the coil etc.

**Reply by Disane87** [reply]
> I have another question, I am currently in the process of making my personal voice assistant, with x integration ranging from opening web pages, to applications, to gpt chat, to video and others, I managed to do  operate the klipper macros by voice command like home, heat, etc., do you have any idea how to macro call this or that coil, eject the coil etc.

I'm actually trying to integrate a button in my integration to change the spools. If I have something I gonna share it with you

**Reply by Eloura74** [reply]
Thanks a lot
